feat(feeds): Commands can now have implicitly use parameters from feed in VM (#664)
unoplatform/uno.extensions: doc/Overview/Reactive/in-apps.md

## Commands

> The generated bindable counterpart of a class will automatically re-expose as `ICommand` public methods that are compatible (cf. below).

Reviewer: Compatible with what ?

Author: Rules are explain "below", but there is no anchor that I can refer to :/

```xml
<Button Command="{Binding Share}" Content="Share" />
```

> 
> By default, if the method has a parameter `T myValue` and there is a property `Feed<T> MyValue` on the same class (matching type and name), 
> that parameter will be automatically full-filed from that feed.

Reviewer: ```suggestion

By default, if the method has a parameter `T myValue` and there is a property `Feed<T> MyValue` on the same class (matching type and name), that parameter will automatically be filled from that feed.
```

For instance, in your ViewModel:

```csharp
public IFeed<string> Message { get; }

public async ValueTask Share(string message) 
{

}
```

Can be used with or without any `CommandParameter` from the view:

```xml
<!-- Without CommandParameter -->
<!-- 'message' arg in the 'Share' method will be the current value of the Message _feed_ -->
<Button Command="{Binding Share}" Content="Share" />

<!-- With CommandParameter -->
<!-- 'message' arg in the 'Share' method will be "hello world" -->
<Button Command="{Binding Share}" CommandParameter="hello world" Content="Share" />
```

> You can even mix "feed parameters" with the "view parameter" (i.e. the value of the `CommandParameter`):

Reviewer: ```suggestion
You can also use both "feed parameters" and "view parameter" (i.e. the value of the `CommandParameter`):
```

```csharp
public IFeed<MyEntity> Entity { get; }

> public async ValueTask Share(MyEntity myEntity, string origin) 

Reviewer: Why isn't the parameter `entity`? How does the matching work?

Author: Humm, indeed the parameter should be `entity`. Matching works on type and name (explained above)

```suggestion
public async ValueTask Share(MyEntity entity, string origin) 
```

{
}
```

```xml
<Button Command="{Binding Share}" CommandParameter="command_bar" Content="Share" />
```

> General rules for you method to be re-exposed as commands are:

Reviewer: ```suggestion
General rules for methods to be re-exposed as commands are:
```

* At most one paramater that cannot be resolved from a _Feed_  property in your VM (a.k.a the `CommandParameter`);
* At most one `CancellationToken`;
* Method can be sync, or async

> [!NOTE]
> The automatic parameters resolution be configured using attributes:
> - `[ImplicitFeedCommandParameters]` on assembly or class to enable or disable the implicit parameters resolution.

> > - `[FeedParameter("MyEntity")]` to explicit the property to use.

Reviewer: This probably needs an explicit example, this is rather important :)
